Experiment_Launcher_code/ExperimentManager.py

```diff
@@ -146,31 +197,40 @@ def StateActivity(self, state, mouse1, mouse2):
             self.mouse_reward = "-"
             self.opponent_reward = "-"
 
-            self.trial_logger.log_trial_data(self.numcompletedtrial, "Decision Abort", self.opponent_choice, self.mouse_choice,self.mouse_reward, self.opponent_reward)
+            self.trial_logger.log_trial_data(self.numcompletedtrial, "Completed Trial", self.opponent_choice,
+                                             self.mouse_choice, self.mouse_reward, self.opponent_reward,
+                                             self.time_start, self.time_to_make_decision, self.time_to_return_to_center)
+
 
         elif state == States.End:
             # Stop recording, finalize logs, show end message, etc.
             self.trial_logger.finalize_logging()
-            analysis_results = self.data_analyzer.analyze_data()
-            print(analysis_results)
+            if self.numcompletedtrial>1:
+                analysis_results = self.data_analyzer.analyze_data()
+                print(analysis_results)
 
     def start_streaming_exp(self, experiment_name, num_trial, decision_time, return_time, opponent_type, opponent1_strategy, opponent2_strategy):
-        self.trial_logger.start_logging(experiment_name)
+
         self.num_trial = num_trial
+
         print("opponent ", opponent_type)
         print("opponent strategy ", opponent1_strategy)
 
         if opponent_type == OpponentType.MOUSE_MOUSE:
             mouse1 = MouseMonitor(1, self.videoAnalyser, self.reward_manager)
             mouse2 = MouseMonitor(2, self.videoAnalyser, self.reward_manager)
+            self.opponent_path ="MOUSE_MOUSE"
         elif opponent_type == OpponentType.MOUSE_COMPUTER:
             mouse1 = MouseMonitor(1, self.videoAnalyser, self.reward_manager)
             mouse2 = FixedStrategyPrisoner(opponent1_strategy, probability=0.8)
+            self.opponent_path = "MOUSE_COMPUTER"
         else:
             mouse1 = FixedStrategyPrisoner(opponent1_strategy, probability=0.8)
             mouse2 = FixedStrategyPrisoner(opponent2_strategy, probability=0.8)
-
-        self.stateManager.SetTimeOut(decision_time, return_time)
+            self.opponent_path = "COMPUTER_COMPUTER"
+        self.trial_logger.start_logging(self.mouse_number,self.opponent_path)
+        force_end_time = num_trial * (decision_time + return_time)
+        self.stateManager.SetTimeOut(decision_time, return_time,force_end_time )
```

Comment:
I did not see the matching change in the StateManager. How come?

Reply:
youre right its not necessary.i must have missed it

Reply:
So, are you fixing it?